# Credit Risk: Supplementary Metric Evidence Audit
This notebook computes metrics from counts in the original report figures. It does not train models, verify the input dataset, reproduce the t-test or recover the original training code. Default (1) is the positive class. This supplemental audit was prepared with AI assistance.


In [1]:
from pathlib import Path
import importlib.util

# Locate the audit module alongside this notebook.
base = Path.cwd()
spec = importlib.util.spec_from_file_location("audit_metrics", base / "audit_metrics.py")
audit = importlib.util.module_from_spec(spec)
spec.loader.exec_module(audit)

for model, counts in audit.MATRICES.items():
    m = audit.calculate(counts)
    print(model)
    print(f"  N={m['test_count']}; TN={m['tn']}; FP={m['fp']}; FN={m['fn']}; TP={m['tp']}")
    print(f"  Accuracy={m['accuracy']:.2%}; default precision={m['default_precision']:.2%}")
    print(f"  Default recall={m['default_recall']:.2%}; F1={m['default_f1']:.2%}")
    print(f"  Balanced accuracy={m['balanced_accuracy']:.2%}")
    print(f"  Test default rate={m['test_default_rate']:.2%}")
    print(f"  Majority non-default accuracy={m['majority_non_default_accuracy']:.2%}")


Logistic regression
  N=5727; TN=4281; FP=232; FN=679; TP=535
  Accuracy=84.09%; default precision=69.75%
  Default recall=44.07%; F1=54.01%
  Balanced accuracy=69.46%
  Test default rate=21.20%
  Majority non-default accuracy=78.80%
Decision tree
  N=5727; TN=4239; FP=274; FN=454; TP=760
  Accuracy=87.29%; default precision=73.50%
  Default recall=62.60%; F1=67.62%
  Balanced accuracy=78.27%
  Test default rate=21.20%
  Majority non-default accuracy=78.80%


## Interpretation
Both models are evaluated on 5,727 observations in the supplied figures. The tree detects 225 more defaults while flagging 42 additional non-defaults. Its default recall is 62.60%, so it still misses 37.40% of observed defaults. No ROC-AUC, PR-AUC or calibration can be recovered from these counts alone. Matching class counts do not independently establish that both matrices used identical borrower rows.

The original code and CSV are required to reproduce training and verify sample membership, cleaning, model settings and statistical assumptions.
